py/usageaccountant/datadog_fetcher.py

```diff
@@ -0,0 +1,309 @@
+import argparse
+import json
+import logging
+import os
+import urllib.parse
+from typing import Any, Mapping, Optional, Sequence, TextIO, Tuple, TypedDict
+from urllib.request import Request, urlopen
+
+from usageaccountant.accumulator import UsageAccumulator, UsageUnit
+
+headers = {
+    "DD-APPLICATION-KEY": os.environ.get("DATADOG_APP_KEY", ""),
+    "DD-API-KEY": os.environ.get("DATADOG_API_KEY", ""),
+}
+
+logger = logging.getLogger("fetcher")
+
+# TODO remove handlers before prod
+logging.basicConfig(
+    level=logging.INFO,
+    format="%(asctime)s [%(levelname)s] %(message)s",
+    handlers=[logging.FileHandler("debug.log"), logging.StreamHandler()],
+)
+
+
+class DatadogResponseUnit(TypedDict):
+    plural: str
+    name: str
+
+
+class DatadogResponseSeries(TypedDict):
+    unit: Sequence[DatadogResponseUnit]
+    pointlist: Sequence[Sequence[float]]
+    scope: str
+
+
+class DatadogResponse(TypedDict):
+    error: Optional[str]
+    query: str
+    from_date: int
+    to_date: int
+    status: str
+    series: Sequence[DatadogResponseSeries]
+
+
+class InvalidResponseError(Exception):
+    def __init__(self, message: str, response: Mapping[Any, Any]) -> None:
+        self.query = response.get("query")
```

Comment:
Please initialize the parent `super().__init__(message)`. Rather than setting the message value here. 
Done this way (this PR) you would be masking any behavior the init method of the parent class could do.